Sources/TuistGraph/DependenciesGraph/ThirdPartyDependency.swift

```diff
@@ -2,20 +2,67 @@ import Foundation
 import TSCBasic
 
 // A enum containing information about third party dependency.
-public enum ThirdPartyDependency: Hashable, Equatable, Codable {
+public enum ThirdPartyDependency: Hashable, Codable {
+    /// A dependency that is imported as source code.
+    case sources(products: [Product], targets: [Target]) // TODO: add the supported platforms read from the SPM package to `sources`
```

Comment:
Should we also add here the minimum deployment target information retrieved from the SPM packages, so that we can perform checks when these are imported?

Reply:
Yes, I think it'd be great if we could make that validation at generation time 👍